packages/nimble-components/src/table/models/tests/sort-operations.spec.ts

```diff
@@ -0,0 +1,99 @@
+import { TableColumnSortOperation } from '../../../table-column/base/types';
+import { getTanStackSortingFunction } from '../sort-operations';
+import type { TableFieldValue } from '../../types';
+
+describe('Sort Operations', () => {
+    function createFakeRow(
+        value: TableFieldValue
+    ): TanStackRow<TableFieldValue> {
+        const row = jasmine.createSpyObj<TanStackRow<TableFieldValue>>(
+            'TanStackRow',
+            ['getValue']
+        );
+        row.getValue.and.returnValue(value);
+        return row;
+    }
+
+    const basicNumericSortOrder = [
+        { value: undefined, name: 'undefined' },
+        { value: null, name: 'null' },
+        { value: Number.NEGATIVE_INFINITY, name: '-∞' },
+        { value: Number.MIN_SAFE_INTEGER, name: 'MIN_SAFE_INTEGER' },
+        { value: -9999, name: '-9999' },
+        { value: -3, name: '-3' },
+        { value: 0, name: '0' },
+        { value: Number.EPSILON, name: 'EPSILON' },
+        { value: 3, name: '3' },
+        { value: 9999, name: '9999' },
+        { value: Number.MAX_SAFE_INTEGER, name: 'MAX_SAFE_INTEGER' },
+        { value: Number.POSITIVE_INFINITY, name: '∞' }
+    ] as const;
+    const basicBooleanSortOrder = [
+        { value: undefined, name: 'undefined' },
+        { value: null, name: 'null' },
+        { value: false, name: 'false' },
+        { value: true, name: 'true' }
+    ] as const;
+    const basicStringSortOrder = [
+        { value: undefined, name: 'undefined' },
+        { value: null, name: 'null' },
+        { value: '', name: "''" },
+        { value: 'abc', name: 'abc' },
+        { value: 'zzz', name: 'zzz' }
+    ] as const;
+
+    const localeAwareCaseSensitiveSortOrder = [
+        { value: undefined, name: 'undefined' },
+        { value: null, name: 'null' },
+        { value: '', name: "''" },
+        { value: 'abc', name: 'abc' },
+        { value: 'ABC', name: 'ABC' },
+        { value: 'hello world', name: 'hello WORLD' },
+        { value: 'xyz', name: 'xyz' }
+    ];
+
+    const sortOrderTestCases = [
+        {
+            sortOperation: TableColumnSortOperation.basic,
+            testCases: [
+                basicNumericSortOrder,
+                basicBooleanSortOrder,
+                basicStringSortOrder
+            ]
+        },
+        {
+            sortOperation: TableColumnSortOperation.localeAwareCaseSensitive,
+            testCases: [localeAwareCaseSensitiveSortOrder]
+        }
+    ] as const;
+    for (const sortOrderTestCase of sortOrderTestCases) {
+        for (const expectedSortOrder of sortOrderTestCase.testCases) {
+            for (let i = 0; i < expectedSortOrder.length; i++) {
+                for (let j = 0; j < expectedSortOrder.length; j++) {
+                    const sortOrderA = expectedSortOrder[i]!;
+                    const sortOrderB = expectedSortOrder[j]!;
+                    it(`comparing ${sortOrderA.name} to ${sortOrderB.name} returns expected value`, () => {
```

Comment:
Could you add more context to the test case names to make them unique (include sort operation and test case name)?

You could also consider rewriting to use the [parameterize utility](https://github.com/ni/nimble/blob/b70ce309d6c03a356b5abaaea7b2199a61f7fbd5/packages/nimble-components/src/utilities/tests/parameterized.ts#L73). This is mostly a suggestion along the lines of "when your favorite tool is a hammer, everything looks like a nail". In fact it may be a bad suggestion. I suspect that would make the tests harder to read in this case: you'd still need just as many loops to generate all the test cases, plus the loop inside `paramaterize` to iterate over them. And since the test cases would be generated rather than static, it might still be hard to `xit` or `fit` on individual cases.

Reply:
I restructured the tests. I'm hoping they are easier to read now. The downside of the restructuring is that now all the `expect` statements for a given sort operation + type is in one test, but I think the readability of the test outweighs the fact that the test name itself doesn't show exactly which comparison caused the test to fail. (Plus, as you pointed out, that wasn't the case before anyway since there were duplicate test names).

Reply:
This format is pretty good. Just to say it, to be on brand completely I'd expect each of the testCases to be separate describe blocks and the names of the specs generated as, for example `undefined equals undefined`:

```ts
describe('Sort Operations', () => {
    describe('basic sorting with numbers', () => {
        // generated via parameterize
        it('undefined equals undefined' , () => {
            // Then either duplicate the test code or parameterize a test function and call it
           // Trying to skip the test structure to avoid a small bit of duplication isn't worth it
        })
    });
});
```

That would uniquify the test names and still be very readable but I don't think it's particularly high value. This PR looks good but it's not a practice I'd advocate generally. If rationalizing maybe it's just for this lots of "mildly" interesting `expect` use cases.

Reply:
I agree that this isn't generally the structure we'd want for our tests. I originally had gone down the path of creating individual tests for the various values we wanted to compare, but it became very difficult very quickly to figure out whether or not I had coverage for all the comparisons we wanted. That's how I landed on having a single array that defined the order and then looping through the array to make sure comparisons between each entry resulted in the correct value. I felt that in the case of these tests, it was better for the tests to more obviously handle all the cases we want to cover even if it required diverging a little from our typical patterns.